### DISCUSSION TITLE
Default settings when upgrading packages

### DISCUSSION THREAD

**Discussion by AnonymousWP** [microsoft/winget-cli#1959] [Q&A] [answered]
### Discussed in https://github.com/microsoft/winget-cli/discussions/1953

<div type='discussions-op-text'>

<sup>Originally posted by **AnonymousWP** February 18, 2022</sup>
Apologies if it has been asked already or if this is the wrong place to ask, but couldn't really find a straightforward answer on the repo nor on Google.

So what are the default settings when I'm upgrading an existing package? For example, when I installed Python the first time, I could check the box that adds Python to PATH, I did not create a start menu shortcut but I did check the box for creating a desktop shortcut. How does Winget handle this? Does it simply check all boxes by default or does it use some configuration file that every application generates after installing? I have found [this](https://www.techwatching.dev/posts/winget-override#:~:text=winget%20install%20official%20documentation%20says,some%20parameters%20for%20the%20program.), but this only mentions how to override and not what the defaults are that Winget uses. Plus, we'd have to look up documentation per application I guess. Hopefully it will be possible to have one repository (or built-in Winget) with all available override options per application.

Besides, is anyone aware of a feature request about changelogs (equivalent to `apt listchanges`)? See: https://manpages.debian.org/testing/apt-listchanges/apt-listchanges.1.en.html.</div>

**Comment by denelon** [comment, marked as answer]
@AnonymousWP,

Different installers have different switches. We primarily automate passing the right switches for "silent", and "silent with progress" for known installer types.

The manifest also has a key to indicate if the previous version should be uninstalled first on upgrade. 

The default behavior is to run the installer in "silent with progress" switches. Any of the settings associated with icons/environment variables/ etc. are dependent on the installer itself.

We don't have a "changelog" to read programmatically with most installers. We've started adding keys to the manifest for things like release dates and release notes, and we can add more, but they aren't at parity with many Linux package distributions yet.

**Reply by AnonymousWP** [reply]
I see, thanks. Any idea how I'd be able to figure out what the 'default' is per installer? As you can probably imagine, updating via Winget create desktop shortcuts again, to name an example.

**Reply by denelon** [reply]
MSIX and MSI installers do follow some standard conventions, but not always. EXE based installers could do almost anything. I would suggest trying some of the scenarios out in a Windows Sandbox so you can observe what happens under different scenarios like installing an earlier version and then running `winget upgrade` to see what happens. That way you aren't cluttering up your desktop or potentially making changes to your primary environment. 

In many cases, the Independent Software Vendor (ISV) or publisher would need to make changes to their installer to improve the behavior. In some cases, the Windows Package Manager might be able to add features. It's always helpful to have reproducible scenarios to validate behavior and test against. There are still plenty of ISVs and publishers who haven't heard about the Windows Package Manager, so they haven't had a chance to determine what the best experience for their customers is.
